src/Symfony/Component/Form/ButtonBuilder.php

```diff
@@ -79,9 +79,9 @@ public function __construct(?string $name, array $options = [])
      *
      * This method should not be invoked.
      *
-     * @param string|int|FormBuilderInterface $child
```

Comment:
All similar change makes sense, but I'm not sure the rest does: passing an int will just work in Symfony 5, thanks to the `string` type. Ppl that use strict mode won't have a smooth upgrade path that's true, but we're going to ask everyone to make their code "strict" while only some actually need it? Not sure...

Reply:
I agree that we should rely on auto casting internally. No need to forbid integers passed from users. See #32066
I don't see what these change try to accomplish.

Reply:
@Tobion , these changes are for the ones calling `$form->add(1)` from classes with strict types enabled. Once we add `string` type in `5.0`, they will have an error.

But now I see the point of @nicolas-grekas better (we already discussed that earlier): if a user does `declare(strict_types=1)`, it's her responsibility to check that she passes correct arguments, because Symfony is not strict about types.

Reply:
Strict types is a bad idea, you have a perfect example of why.
It provides no additional type-guarantee.
On the contrary, type declarations on signature boundaries *do* provide additional guarantee and thus prevent bugs.
Good news, we're adding them all in v5 when technically possible :)

Reply:
See #32828 as alternative.

Reply:
@nicolas-grekas , thank toy, but I could have done the same in this PR by just removing the deprecations... I raised the issue to discuss it. Waiting for @xabbuh comment, who also participated in this PR.

Reply:
IMO being nice to developers using strict types doesn't hurt and saves them some pain when upgrading. That's why I would be in favour of merging this PR (though we can also merge #32828).

Reply:
But being nice to some means being not-nice to others...

Reply:
Whom are we not being nice when adding deprecations? :)

Reply:
What I meant above:

> are we going to ask everyone to make their code "strict" while only some actually need it?

this deprecation will be shown to people that don't use strict types, and they will have to add casts to string for nothing.

Reply:
To summarize, there are two options:
1. deprecate passing ints and ask everyone to cast to string,
2. do not deprecate ints and only strict type users will have to cast to string.

It feels like I would prefer the second alternative between the two. After all strict type users should be aware that the 3rd party code they are calling might be not strictly typed.

Reply:
If we go for 2., we will have to add such deprecations everywhere in the whole codebase. No-go.

Reply:
@nicolas-grekas , you mean 1? :)
@xabbuh, so, do you agree that we should close this?

Reply:
I mean that this situation will happen to all places where we added the `string` type - not only on forms.

Reply:
I would have only done that in places where we explicitly declared support for other data types before. But I also understand the argument about everyone having to do the type cast then (though I wouldn't expect that many people actually use integers as form names).

Reply:
Yes, I understand. Should we add a note in our BC promise about that?

Reply:
Sure, I meant 1. :)
The BC promise is only about minor versions, so that's a bit off topic, but I don't have a better proposal.
PR welcome on the BC promise then, to be clear we don't promise not hard-breaking strict types.